sdk/src/instruction_processor_utils.rs

```diff
@@ -27,6 +27,29 @@ macro_rules! solana_entrypoint(
     )
 );
 
+#[macro_export]
+macro_rules! solana_program(
+    ($id:ident, $bs58:expr, $name:ident, $entrypoint:ident) => (
+        $crate::solana_name_id!($id, $bs58);
+
+        #[macro_export]
+        macro_rules! $name {
+            () => {
+                (stringify!($name).to_string(), $crate::id())
+            };
+        }
+
+        #[no_mangle]
+        pub extern "C" fn $name(
```

Comment:
This should probably include a suffix like `_entrypoint`. Then `$name` can be just one word, such as `stake` or `config`.

Reply:
`name` is also used for the filename, we might want to keep the long name like `solana_stake_program`.  The particular naming of the entrypoint is pretty much invisible to everyone but the native loader.  Using the package name would be ideal.

Reply:
Could use:
```
const NAME: &'static str = env!("CARGO_PKG_NAME");
```

But would have to lower dashes to underlines